src/AspNetCoreModuleV2/CommonLib/config_utility.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <map>
+
+class ConfigUtility
+{
+    #define CS_ASPNETCORE_HANDLER_SETTINGS                   L"handlerSettings"
+    #define CS_ASPNETCORE_HANDLER_VERSION                    L"handlerVersion"
+    #define CS_ASPNETCORE_HANDLER_SETTINGS_NAME              L"name"
+    #define CS_ASPNETCORE_HANDLER_SETTINGS_VALUE             L"value"
+
+public:
+    static
+    HRESULT
+    FindHandlerVersion(IAppHostElement* pElement, STRU* strHandlerVersionValue)
+    {
+        HRESULT hr = S_OK;
+        IAppHostElement                *pHandlerSettings = NULL;
+        IAppHostElementCollection      *pHandlerSettingsCollection = NULL;
+        ENUM_INDEX                      index;
+        IAppHostElement                *pHandlerVar = NULL;
+        STRU strHandlerName;
+        STRU strHandlerValue;
+
+        hr = GetElementChildByName(pElement,
+            CS_ASPNETCORE_HANDLER_SETTINGS,
```

Comment:
please call release for pHandlerSettings , pHandlerSettingsCollection  and pHandlerVar, otherwise memory leak

Reply:
Or stick them into unique_ptr.

Reply:
As this is used by the aspnetcore_shim_config, I'm going to avoid std for now. I'd rather have the class all or none for exception handling.

Reply:
Right, but the pattern we use here (setting the field) is not sharable anyway. Additionally, they would probably care about different properties.

Reply:
Good point. I originally implemented this to populate a std::map.